### PR TITLE
Warning removal from hypervisor codebase

### PR DESCRIPTION
**Pull request by kbillore**
This patch prevents direct usage of void pointer without cast.
Tracked-On: #2955
Signed-off-by: Kaushal Billore <kaushal.billore@intel.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: hypervisor/boot/sbl/multiboot.c [hunk 1]
```diff
@@ -54,7 +54,7 @@ static void parse_other_modules(struct acrn_vm *vm, const struct multiboot_modul
 			static int32_t copy_once = 1;
 
 			start = end + 1; /*it is fw name for boot args */
-			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%x ", start, mod_size, mod_addr);
+			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%x ", start, mod_size, (uint64_t)mod_addr);
```
Comment (dongyaozu) [inline hypervisor/boot/sbl/multiboot.c]: are  u using old code base? I didn't see this

Comment (kbillore) [reply hypervisor/boot/sbl/multiboot.c]: I am using master branch. (git clone https://github.com/projectacrn/acrn-hypervisor.git)

**Comment by acrnsi** [comment]
Ok to verify

File: hypervisor/boot/sbl/multiboot.c (hunk 1 above)
Comment (dongyaozu) [reply hypervisor/boot/sbl/multiboot.c]: Then, should we use %p？ or, if we use mod_addr, should we use %lx ?

**Comment by HuiHuangShi** [comment]
out of time

**Comment by kbillore** [comment]
> out of time

means?

**Comment by HuiHuangShi** [comment]
> > out of time
> 
> means?

Use code script to scan the pull reuqest, and the time is limited, the script is out of time, and will rescan again, will provide the report in the github comment,thanks.

**Comment by HuiHuangShi** [comment]
1 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/boot/sbl/multiboot.c func:parse_other_modules offset:57<br> reason:Cast from pointer to integral type. : (void* to unsigned long): ( uint64_t ) mod_addr<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
> 1 new potential violations to the coding guideline detected. Please help check. Thanks!
> <1> config:common filename:/hypervisor/boot/sbl/multiboot.c func:parse_other_modules offset:57
> reason:Cast from pointer to integral type. : (void* to unsigned long): ( uint64_t ) mod_addr
> 
> No New Name Conflict
> 
> No New Function Declaration/Definition Mismatch

@kbillore can you fix it? Do not use type conversion here, and pls use %p

**Comment by kbillore** [comment]
> > 1 new potential violations to the coding guideline detected. Please help check. Thanks!
> > <1> config:common filename:/hypervisor/boot/sbl/multiboot.c func:parse_other_modules offset:57
> > reason:Cast from pointer to integral type. : (void* to unsigned long): ( uint64_t ) mod_addr
> > No New Name Conflict
> > No New Function Declaration/Definition Mismatch
> 
> @kbillore can you fix it? Do not use type conversion here, and pls use %p

done!

File: hypervisor/boot/sbl/multiboot.c (hunk 1 above)
Comment (kbillore) [reply hypervisor/boot/sbl/multiboot.c]: @dongyaozu used %p, thats better.

**Comment by kbillore** [comment]
diff --git a/hypervisor/boot/sbl/multiboot.c b/hypervisor/boot/sbl/multiboot.c
index e0758796b..c7bb0fff8 100644
--- a/hypervisor/boot/sbl/multiboot.c
+++ b/hypervisor/boot/sbl/multiboot.c
@@ -54,7 +54,7 @@ static void parse_other_modules(struct acrn_vm *vm, const struct multiboot_modul
 			static int32_t copy_once = 1;
 
 			start = end + 1; /*it is fw name for boot args */
-			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%x ", start, mod_size, mod_addr);
+			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%p ", start, mod_size, mod_addr);
 			dev_dbg(ACRN_DBG_BOOT, "fw-%d: %s", i, dyn_bootargs);
 
 			/*copy boot args to load addr, set src=load addr*/

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by kbillore** [comment]
> No new violations to the coding guideline detected.
> No New Name Conflict
> 
> No New Function Declaration/Definition Mismatch

How to clear this?

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by dongyaozu** [comment]
@binbinwu1 can u review?

File: hypervisor/boot/sbl/multiboot.c [hunk 2]
```diff
@@ -54,7 +54,7 @@ static void parse_other_modules(struct acrn_vm *vm, const struct multiboot_modul
 			static int32_t copy_once = 1;
 
 			start = end + 1; /*it is fw name for boot args */
-			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%x ", start, mod_size, mod_addr);
+			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%p ", start, mod_size, mod_addr);
```
Comment (tw4452852) [inline hypervisor/boot/sbl/multiboot.c]: As our implementation of `snprintf` is [here](https://github.com/projectacrn/acrn-hypervisor/blob/948d58fb9c0b30bb4cab72c071e86cd3d0cce01c/hypervisor/lib/sprintf.c#L415), I don't think we support `%p` for now.

Comment (binbinwu1) [reply hypervisor/boot/sbl/multiboot.c]: yes, it is not supported, we need to document the supported formation more clearly...

File: _config.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+theme: jekyll-theme-cayman
```
Comment (binbinwu1) [inline _config.yml]: What's the purpose by adding this?

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
> > No new violations to the coding guideline detected.
> > No New Name Conflict
> > No New Function Declaration/Definition Mismatch
> 
> How to clear this?

It means there is no violation in your pull request, scan script pass the pull request.

**Comment by acrnsi** [comment]
@kbillore hi ,please add 
“
Tracked-On: #2957
Signed-off-by: Kaushal Billore <kaushal.billore@intel.com>
”
 all of your  commits

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request closed without merging**